tools/ci_build/build.py

```diff
@@ -1484,27 +1484,29 @@ def generate_build_tree(
                     f"-DVERSION_PRIVATE_PART={MM}{DD}",
                     f"-DVERSION_STRING={ort_major}.{ort_minor}.{build_number}.{source_version[0:7]}",
                 ]
-    cflags = None
-    cxxflags = None
-    ldflags = None
-    cudaflags = []
+
     for config in configs:
+        cflags = []
+        cxxflags = None
+        ldflags = None
+        cudaflags = []
+        if is_windows() and not args.ios and not args.android and not args.build_wasm:
```

Comment:
maybe longer term we can find a better way to determine what compiler is being used. I wonder if detecting a VS generator would be sufficient? also, `is_windows() and args.ios` is not actually supported right?

fine with this for now though.

Reply:
Yes, we should have such a helper function.